packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -101,6 +101,17 @@ function build(
   const safetyPins = new Set(); // temporary
   const deadSet = new Set(); // vrefs that are finalized but not yet reported
 
+  function retainExportedRemotable(vref) {
+    // if the vref corresponds to a Remotable, keep a strong reference to it
+    // until the kernel tells us to release it
+    const { type, allocatedByVat, virtual } = parseVatSlot(vref);
+    if (type === 'object' && allocatedByVat && !virtual) {
+      const remotable = slotToVal.get(vref).deref();
+      assert(remotable, X`somehow lost Remotable for ${vref}`);
```

Comment:
I presume this could happen between serialization and when this is called, but when I look at the code below I'm convinced that value will always be on the stack in the six cases we have.  However, it's probably worth a comment somewhere mentioning that this is now an invariant we need to preserve.

Reply:
yeah, I agree that this should never happen